### DISCUSSION TITLE
Config to swap A<->B buttons

### DISCUSSION THREAD

**Discussion by EvilToasterDBU** [diasurgical/DevilutionX#5506] [Ideas]
My controller is fine defined and working. But the controller is only for android and it is not possible to connect it to a PC to create a mapping. It would be great to be able to swap A and B buttons without having to generate a mapping.

If it matters, I use an gamesir x2 pro xbox controller.

**Comment by AJenbo** [comment]
remapping will be supported in 1.5.0

**Comment by rtrzebinski** [comment]
@EvilToasterDBU Please check https://github.com/diasurgical/devilutionX/wiki/Game-Controller-Scheme#how-to-change-the-default-mapping-case-study---update-ps5-controller-so-a--b----b--a----this-will-become-more-natural-for-ps-users

**Reply by AJenbo** [reply]
in 1.5.0 it also detects what type of gamepad you have instead of only doing things in the nintendo style. i would suggest you checkout the beta so you can se if it works correctly and give feedback before release

**Reply by rtrzebinski** [reply]
I've just checked the latest built - didn't work very well, wrong keys were detected, perhaps because I've already has a modified mapping in diablo.ini?

Anyway I'm happy to help with testing, playing with the controller is (surprisingly) really awesome :)

**Reply by EvilToasterDBU** [reply]
I think that the main gamepad layout should be adjusted to control as on PS1, it seems to me it will be more convenient. The correct layout for the menu is good, but a completely different layout for the gameplay is strange in my opinion.

**Reply by AJenbo** [reply]
@rtrzebinski Yeah overwriting the mapping effectively swaps the buttons so that would conflict with automatic handling.

@EvilToasterDBU convenient for someone who is more used to the PS1 layout maybe, but that isn't much of an argument.

**Reply by EvilToasterDBU** [reply]
@AJenbo okay, here's an example: the player presses the talk to the character button, usually this button is the same as the action confirmation button in the dialog menu (in other games for example), which is logical, but in our case it turns out not so

**Reply by rtrzebinski** [reply]
I like the idea of defaulting to PS1 mapping once PS4/5 controller was connected, with the menu option to change it anyhow.

**Reply by AJenbo** [reply]
@EvilToasterDBU Please open an issue regarding the inconsistent button action

**Reply by StephenCWills** [reply]
We can just use #5467 rather than opening a new issue.
